ecs-cli/modules/utils/compose/convert_task_definition.go

```diff
@@ -602,3 +629,11 @@ func SortedGoString(v interface{}) (string, error) {
 	}
 	return string(b), nil
 }
+
+func areContainersAllEssential(ecsContainerDefs ContainerDefs) bool {
+	// NOTE: This is a silly hack for now, since the only field in a
+	// ContainerDef is essential. If/when there are other fields, we would
+	// want to change the logic for determining whether the essential field
+	// in all Containers was specified.
+	return len(ecsContainerDefs) == 0
```

Comment:
Looking at this made me wonder if there is an edge case that we haven't considered. What if you have an ECS Params file like this:

```
task_definition:
  task_role_arn: string
  ecs_network_mode: string # valid values: "none", "host", "bridge", and custom ECS network modes.
  services:
    mysql:
    wordpress:
    redis:
```

This should probably be an error, but I have a suspicion the YAML library my parse it and not throw an error- and that this will cause odd behavior in your code.

Reply:
This will actually set "essential" to false for all those services. Tests now reflect that. Not sure if that is desirable behavior, but it doesn't throw a YAML parsing error.

Reply:
Okay, I think that's acceptable. I mostly just wanted to make sure this wouldn't cause your code to get a null pointer exception or something like that.